src/main/java/com/aws/greengrass/security/SecurityService.java

```diff
@@ -89,4 +113,51 @@ private CryptoKeySpi selectCryptoKeyProvider(String keyUri) throws URISyntaxExce
         }
         return provider;
     }
+
+    static class DefaultCryptoKeyProvider implements CryptoKeySpi {
+        private static final Logger logger = LogManager.getLogger(DefaultCryptoKeyProvider.class);
+
+        @Override
+        public KeyManager[] getKeyManagers(String privateKeyUriStr, String certificateUriStr)
+                throws KeyLoadingException, URISyntaxException {
+            URI privateKeyUri = new URI(privateKeyUriStr);
+            if (!isUriSupportedKeyType(privateKeyUri)) {
+                logger.atError().kv(KEY_URI, privateKeyUriStr).log("Can't process the key type");
+                throw new KeyLoadingException(String.format("Only support %s type private key", supportedKeyType()));
+            }
+
+            URI certificateUri = new URI(certificateUriStr);
+            if (!isUriSupportedKeyType(certificateUri)) {
+                logger.atError().kv(CERT_URI, certificateUriStr).log("Can't process the certificate type");
+                throw new KeyLoadingException(String.format("Only support %s type certificate", supportedKeyType()));
+            }
+
+            try {
+                PrivateKey privateKey = EncryptionUtils.loadPrivateKey(privateKeyUri.getPath());
+                List<X509Certificate> certificateChain = EncryptionUtils.loadX509Certificates(certificateUri.getPath());
+
+                KeyStore keyStore = KeyStore.getInstance("PKCS12");
+                keyStore.load(null);
+                keyStore.setKeyEntry("private-key", privateKey, null, certificateChain.toArray(new Certificate[0]));
+
+                KeyManagerFactory keyManagerFactory =
+                        KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+                keyManagerFactory.init(keyStore, null);
+                return keyManagerFactory.getKeyManagers();
+            } catch (GeneralSecurityException | IOException e) {
+                logger.atError().kv(KEY_URI, privateKeyUriStr).kv(CERT_URI, certificateUriStr)
+                        .log("Exception caught during getting key manager");
+                throw new KeyLoadingException("Failed to get key manager", e);
+            }
+        }
+
+        @Override
+        public String supportedKeyType() {
+            return "file";
```

Comment:
should this be a list/enum somewhere?

Reply:
+1 for enum

Reply:
Because this is a public interface, a enumerable finite list is not applicable I think. Because customer could define its own key type. But I definitely could change it to a constant.